src/App.tsx

```diff
@@ -1,61 +1,137 @@
+    ...todo,
+    user: findUserById(todo.userId),
+  };
+});
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState('');
+  const [todos, setTodos] = useState(preparedTodos);
+  const [isErrorOnUserSelect, setErrorOnUserSelect] = useState(false);
+  const [isErrorOnTitleInput, setErrorOnTitleInput] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.currentTarget.value);
+    setErrorOnTitleInput(false);
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUserId(event.currentTarget.value);
+    setErrorOnUserSelect(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setErrorOnTitleInput(!title.trim());
+    setErrorOnUserSelect(!selectedUserId);
+
+    if (title.trim() === '' || !selectedUserId) {
+      return;
+    }
+
+    const userToAdd = findUserByName(selectedUserId);
+
+    setTodos(current => {
+      const maxTodoId = Math.max(...current.map(todo => todo.id));
+
+      const todoItem = {
+        id: maxTodoId + 1,
+        title,
+        completed: false,
+        userId: userToAdd ? userToAdd.id : null,
+        user: userToAdd,
+      };
+
+      return [
+        ...current,
+        todoItem,
+      ];
+    });
+
+    setTitle('');
+    setSelectedUserId('');
+  };
+
   return (
     <div className="App">
-      <h1>Add todo form</h1>
+      <h1 className="title is-1">Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form className="form" onSubmit={handleSubmit}>
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+
+          <input
+            data-cy="titleInput"
+            type="text"
+            id="title"
+            name="title"
+            value={title}
+            className="input is-rounded"
+            placeholder="Enter title of task"
+            onChange={handleTitleChange}
+          />
+
+          {isErrorOnTitleInput && (
+            <span className="error">Please enter a title</span>
+          )}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
+          <label
+            htmlFor="userSelect"
+            className="label"
+          >
+            Choose user:
+          </label>
+
+          <select
+            data-cy="userSelect"
+            id="userSelect"
+            name="userSelect"
+            value={selectedUserId}
+            className="input is-rounded"
+            onChange={handleUserChange}
+          >
+            <option value="" disabled>Choose a user</option>
+
+            {usersFromServer.map(user => (
+              <option key={user.id} value={user.name}>{user.name}</option>
```

Comment:
It would be best to use userId as a value for the MenuItem. You don't need two functions to find the user (by name and by id), but only one (by id)

Reply:
fixed

Reply:
I hipe I understand you correctly, fixed it